src/transformers/models/qwen3_moe/modeling_qwen3_moe.py

```diff
@@ -266,7 +266,8 @@ def forward(self, hidden_states: torch.Tensor) -> torch.Tensor:
         expert_mask = torch.nn.functional.one_hot(selected_experts, num_classes=self.num_experts).permute(2, 1, 0)
 
         # Loop over all available experts in the model and perform the computation on each expert
-        for expert_idx in range(self.num_experts):
+        expert_hitted = (expert_mask.sum(dim=(-1, -2)) > 0).nonzero(as_tuple=True)[0].tolist()
```

Comment:
BTW i checked and we don't need the `.tolist()` 
```suggestion
        expert_hitted = torch.greater(expert_mask.sum(dim=(-1, -2)), 0).nonzero()
```
should work. 
Can you update the other models as well.